ids/node_id.go

```diff
@@ -4,51 +4,57 @@
 package ids
 
 import (
-	"bytes"
-	"errors"
 	"fmt"
+	"strings"
 
 	"github.com/ava-labs/avalanchego/staking"
 	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/cb58"
 	"github.com/ava-labs/avalanchego/utils/hashing"
 )
 
 const (
 	NodeIDPrefix = "NodeID-"
-	NodeIDLen    = ShortIDLen
+
+	LongNodeIDLen = 32
 )
 
 var (
 	EmptyNodeID = NodeID{}
 
-	errShortNodeID = errors.New("insufficient NodeID length")
-
-	_ utils.Sortable[NodeID] = NodeID{}
+	_ utils.Sortable[NodeID] = (*NodeID)(nil)
 )
 
-type NodeID ShortID
-
-func (id NodeID) String() string {
-	return ShortID(id).PrefixedString(NodeIDPrefix)
+// NodeID embeds a string, rather than being a type alias for a string
+// to be able to use custom marshaller for json encoding.
+// See https://github.com/golang/go/blob/go1.20.8/src/encoding/json/encode.go#L1004-L1026
+// which checks for the string type first, then checks to see if a custom marshaller exists,
+// then checks if any other of the primitive types are provided.
+type NodeID struct {
```

Comment:
@StephenButtolph with the string embedded, we could just use pointer receivers instead of copying NodeID around. Any objection?

Reply:
`string`s are already passed by reference (I believe) - so I don't think we really need to worry about this